wordpress-mobile/WordPress-iOS
# Issues/login jetpack discovery

This PR adds a check for Jetpack to the end of the XML-RPC discovery process.  If we're unable to find the XML-RPC endpoint of a Jetpack connected blog, this look up will allow us to direct the user to Jetpack support to resolve the issue.
Note that this PR leaves intact the error override in `[WordPressXMLRPCAPIFacade guessXMLRPCURLForSite:success:failure]` and error handling will be updated in a subsequent PR.

To test:
You'll want a self-hosted blog to test with. 
You'll want to add a breakpoint in either `WordPressXMLRPCAPIFacade` before the error is over written, or in `WordPressOrgXMLRPCValidator` so you can inspect the error returned.

Scenario 0: 
Make sure you can login to your self-hosted site.

Scenario 1:
Prep: Edit your xmlrpc.php so it throws a PHP error.
Attempt to sign into your self-hosted site. 
Confirm the error indicates Jetpack is inaccessable.

Scenario 2:
Prep:
Edit your .htaccess to restrict access to xmlrpc.php.
Attempt to sign into your self-hosted site. 
Confirm the error indicates Jetpack is inaccessable.

Scenario 3: 
Prep: Install Jetpack but ensure it is disconnected.
Attempt to sign into your self-hosted site.
Confirm the error indicates Jetpack is disconnected.

Needs review: @koke would you do the honors?

cc @nheagy
